core/test/math/bigint_tests.cpp

```diff
@@ -176,4 +177,16 @@ TEST(BigInt, FromFloatString) {
     auto t = witness.toByteArray();
     auto tt = subject.toByteArray();
     EXPECT_TRUE(witness.compare(subject) == 0);
+}
+
+TEST(BigInt, AssignFromUnsigned64bitScalar) {
+    auto value = std::numeric_limits<uint64_t>::max();
+    BigInt bigInt = BigInt::fromScalar(value);
+    EXPECT_EQ(value, bigInt.toUint64());
+}
+
+TEST(BigInt, AssignFromSigned64bitScalar) {
+    uint64_t value = std::numeric_limits<uint64_t>::min();
+    BigInt bigInt = BigInt::fromScalar(value);
+    EXPECT_EQ(value, bigInt.toUint64());
```

Comment:
Why use `BigInt::fromScalar` instead of `assignScalar` ? I suggest to be as close as possible of the tested method/function in unit test.